src/transformStateWithClones.js

```diff
@@ -7,7 +7,26 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let newState = { ...state };
+  const states = [];
+
+  for (const action of actions) {
+    if (action.type === 'addProperties') {
+      newState = { ...newState, ...action.extraData };
+    } else if (action.type === 'removeProperties') {
+      const updatedState = { ...newState };
+
+      for (const keyToRemove of action.keysToRemove) {
+        delete updatedState[keyToRemove];
+      }
+      newState = updatedState;
+    } else if (action.type === 'clear') {
+      newState = {};
+    }
```

Comment:
A default case should be added in the switch statement for error handling, to account for cases where the 'type' property does not match any of the known actions.